sql/tablewriter.go

```diff
@@ -641,39 +639,39 @@ func (td *tableDeleter) deleteAllRowsScan() error {
 			// Done deleting rows.
 			break
 		}
-		_, err = td.row(row)
+		_, err = td.row(ctx, row)
 		if err != nil {
 			return err
 		}
 	}
-	return td.finalize()
+	return td.finalize(ctx)
 }
 
 // deleteIndex runs the kv operations necessary to delete all kv entries in the
 // given index. This may require a scan.
-func (td *tableDeleter) deleteIndex(idx *sqlbase.IndexDescriptor) error {
+func (td *tableDeleter) deleteIndex(ctx context.Context, idx *sqlbase.IndexDescriptor) error {
 	if len(idx.Interleave.Ancestors) > 0 || len(idx.InterleavedBy) > 0 {
 		if log.V(2) {
-			log.Info(context.TODO(), "delete forced to scan: table is interleaved")
+			log.Info(ctx, "delete forced to scan: table is interleaved")
 		}
-		return td.deleteIndexScan(idx)
+		return td.deleteIndexScan(ctx, idx)
 	}
-	return td.deleteIndexFast(idx)
+	return td.deleteIndexFast(ctx, idx)
 }
 
-func (td *tableDeleter) deleteIndexFast(idx *sqlbase.IndexDescriptor) error {
+func (td *tableDeleter) deleteIndexFast(ctx context.Context, idx *sqlbase.IndexDescriptor) error {
 	indexPrefix := sqlbase.MakeIndexKeyPrefix(td.rd.helper.tableDesc, idx.ID)
 	indexStartKey := roachpb.Key(indexPrefix)
 	indexEndKey := indexStartKey.PrefixEnd()
 
 	if log.V(2) {
-		log.Infof(context.TODO(), "DelRange %s - %s", indexStartKey, indexEndKey)
+		log.Infof(ctx, "DelRange %s - %s", indexStartKey, indexEndKey)
```

Comment:
didn't you file an issue to centralize all this logging logic? might be worth doing it now.